### PR TITLE
SpectrogramSequence

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR address Issue #142.  It renames the ```Raster``` class to ```SpectrogrogramCube```, implements a new class, ```SpectrogrogramSequence```, for sequences of ```SpectrogrogramCube```s that don't represent a raster scan, and refactors ```RasterSequence``` to inherit from ```SpectrogrogramSequence```.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
ping @hayesla.  This PR addresses what we discussed yesterday.  Any thoughts?  No need to fork it and play with it at this stage.  More high level thoughts (or obvious coding mistakes) would be helpful at this stage.

File: sunraster/spectrogram_sequence.py [hunk 1]
```diff
@@ -17,97 +18,32 @@
...
 
-class RasterSequence(NDCubeSequence):
+class SpectrogramSequence(NDCubeSequence, SpectrogramABC):
     """
-    Class for holding, slicing and plotting spectrogram data.
+    Class for holding, slicing and plotting a sequence of spectrogram cubes.
 
-    This class contains all the functionality of its super class with
-    some additional functionalities.
+    Spectrogram cubes can be 2D or higher.
```
Comment (hayesla) [inline sunraster/spectrogram_sequence.py:25]: how high? can it not only go to three?

File: sunraster/spectrogram.py [hunk 2]
```diff
@@ -49,7 +50,59 @@
 SUPPORTED_EXPOSURE_NAMES = np.array(SUPPORTED_EXPOSURE_NAMES)
 
 
-class Raster(NDCube):
+class SpectrogramABC(abc.ABC):
+
```
Comment (hayesla) [inline sunraster/spectrogram.py:54]: maybe add a docstring here just to help describe this a bit better?:)

Comment (DanRyanIrish) [inline sunraster/spectrogram.py:54]:
```suggestion
    """Abstract Base Class to define the base API of Spectrogram objects."""
```

File: sunraster/spectrogram_sequence.py (hunk 1 above)
Comment (DanRyanIrish) [reply sunraster/spectrogram_sequence.py:25]: I don't actually think ```SpectrogramCube``` or ```SpectrogramSequence``` enforce an upper limit of dimensions, just so long as the expected dimensions are somewhere it there...

File: sunraster/spectrogram.py (hunk 2 above)
Comment (DanRyanIrish) [reply sunraster/spectrogram.py:54]: Dont wan't to end up in any confusing docstring inheritance situations.  ```NDCubeABC``` doesn't have a docstring.  I'll put a comment above the class instead explain what it is.

Comment (DanRyanIrish) [inline sunraster/spectrogram.py:54]:
```suggestion

    # Abstract Base Class to define the basic API of Spectrogram classes.

```

**Pull request closed without merging**